ui/src/app/guards/disabled-guard.service.ts

```diff
@@ -22,7 +19,7 @@ export class DisabledGuard implements CanActivate, CanActivateChild {
       if (!isSignedIn) {
         return false;
       }
-      await this.profileStorageService.profile$.first().toPromise();
+      await profileStore.get().reload();
```

Comment:
This is different from what the original was doing: the previous approach just blocked on the initial value to load - this appears to refetch the profile every time this guard runs (I don't know how often this is). Does this need to call reload? If yes: I would navigate around locally and verify what network traffic you observe for GET profile (you should not see new requests on every page change).

Reply:
I did this because the catch block looking for an API error made me think that this was intended to catch a user being disabled in the middle of using the app. Possible that that's just not what we're going for and I should figure out some other way to handle disabled users.

Reply:
Right - the catch was being used to detect whether you are disabled, but it only ran on the cached result in the storage service. The previous code was not triggering a new request on every guard run. We don't want to poll on every navigate to ask if the user has been disabled yet, since it's a very rare scenario, and the user is probably doing something bad/suspect anyways if they are disabled mid-use. They'll start seeing seemingly random errors from most endpoints (403), eventually refresh, then they will hit this guard again on page reload.

The behavior we want here is:
- block until we know any recent value for the profile (anytime within this client session is fine)
- if instead there's an error when loading the profile, handle it per previous code

This will depends on how the `store` deals with initial load errors, and how it supports client who want a "lazy get"-like behavior.

Reply:
[76bd871 ](https://github.com/all-of-us/workbench/pull/4934/commits/76bd871ba5b2d7536e9e63e1d6c5f3bba7496558) updates the store to support a lazy get.